packages/dds/matrix/src/matrix.ts

```diff
@@ -472,10 +472,17 @@ export class SharedMatrix<T = any>
 	 * @param callback - code that needs to protected against reentrancy.
 	 */
 	private protectAgainstReentrancy(callback: () => void) {
+		if (this.reentrantCount !== 0) {
+			// Validate that applications don't submit edits in response to matrix change notifications. This is unsupported.
+			throw new UsageError("Reentrancy detected in SharedMatrix.");
+		}
 		assert(this.reentrantCount === 0, 0x85d /* reentrant code */);
```

Comment:
```suggestion
		assert(this.reentrantCount === 0, 0x85d /* reentrant code */);
```
we can remove this as the above line already ensures the condition. 

should the final assert also be a usage error?

Reply:
oops, yeah.

I think the final assert is ok to leave as an assert--it's a consistency check that the scheme for detecting reentrancy is correct. But I'd also be ok removing it.

Reply:
ahh. ok. makes sense. maybe update the comment text to describe what the alert catches